### PR TITLE
Create test for gray scale transformation

### PR DESCRIPTION
**Pull request by jeziellago**
New test for gray scale transformation, asserting that it transform correctly.

### CODE REVIEW

File: coil-base/src/androidTest/java/coil/transform/GrayscaleTransformationTest.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        val coloredBitmap = getColoredBitmap()
+
+        val grayscaleBitmap = runBlocking {
+            grayscaleTransformation.transform(pool, coloredBitmap)
+        }
+
+        val (_, red, green, blue) = grayscaleBitmap.getPixels()
+
+        assertTrue(red.contentEquals(green) && green.contentEquals(blue))
```
Comment (colinrtwhite) [inline coil-base/src/androidTest/java/coil/transform/GrayscaleTransformationTest.kt]: To make this test more robust, I would add the output of the grayscale transformation to the test assets as `normal_grayscale.jpg`. Then you can use [Bitmaps.sameAs(other, 0.99)](https://github.com/coil-kt/coil/blob/master/coil-base/src/sharedTest/java/coil/util/Bitmaps.kt#L39) to verify that the result matches the expected output.

File: coil-base/src/androidTest/java/coil/transform/GrayscaleTransformationTest.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        assertFalse(grayscaleBitmap.sameAs(coloredBitmap))
+    }
+
+    private fun getColoredBitmap(): Bitmap {
```
Comment (colinrtwhite) [inline coil-base/src/androidTest/java/coil/transform/GrayscaleTransformationTest.kt]: ^ Can remove this function after moving to [Bitmaps.sameAs(other, 0.99)](https://github.com/coil-kt/coil/blob/master/coil-base/src/sharedTest/java/coil/util/Bitmaps.kt#L39).

**Comment by colinrtwhite** [comment]
Thanks for the contribution! Left a few comments.

File: coil-base/src/androidTest/java/coil/transform/GrayscaleTransformationTest.kt (hunk 1 above)
Comment (jeziellago) [reply coil-base/src/androidTest/java/coil/transform/GrayscaleTransformationTest.kt]: Nice! I agree you! Can you review if make sense my last changes? Thanks! :)

**Comment by jeziellago** [comment]
@colinrtwhite Can you review if make sense my last changes? Thanks! :)

File: coil-base/src/androidTest/java/coil/transform/GrayscaleTransformationTest.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        assertTrue(red.contentEquals(green) && green.contentEquals(blue))
+        assertTrue(grayscaleBitmap.sameAs(normalGrayscaleBitmap, THRESHOLD))
+    }
+
+    private fun getBitmapFromAssets(
+        fileName: String
+    ): Bitmap {
+        val source = context.assets.open(fileName).source().buffer()
+        val (drawable, _) = runBlocking {
+            decoder.decode(
```
Comment (colinrtwhite) [inline coil-base/src/androidTest/java/coil/transform/GrayscaleTransformationTest.kt]: To keep this unit test minimal, I would not use `BitmapFactoryDecoder` in this test. Instead, I would remove this method and inline `BitmapFactory.fromInputStream(context.assets.open(fileName))`.

Comment (jeziellago) [reply coil-base/src/androidTest/java/coil/transform/GrayscaleTransformationTest.kt]: Make sense!

**Comment by jeziellago** [comment]
@colinrtwhite Done!

**Pull request merged**